File: src/extras-kwidgetsaddons/libkstyleextensions.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const qstyle_enums = @import("../libqstyle.zig").enums;

/// https://api-staging.kde.org/kstyleextensions.html
pub const kstyleextensions = struct {
    /// [Qt documentation](https://api-staging.kde.org/kstyleextensions.html#customControlElement)
    ///
    /// ``` param1: []const u8, param2: QtC.QWidget ```
    ///
    /// Returns: ``` qstyle_enums.ControlElement ```
    pub fn CustomControlElement(param1: []const u8, param2: ?*anyopaque) i64 {
        const param1_str = qtc.libqt_string{
            .len = param1.len,
            .data = param1.ptr,
        };
        return qtc.KStyleExtensions_CustomControlElement(param1_str, @ptrCast(param2));
    }

    /// [Qt documentation](https://api-staging.kde.org/kstyleextensions.html#customStyleHint)
    ///
    /// ``` param1: []const u8, param2: QtC.QWidget ```
    ///
    /// Returns: ``` qstyle_enums.StyleHint ```
    pub fn CustomStyleHint(param1: []const u8, param2: ?*anyopaque) i64 {
        const param1_str = qtc.libqt_string{
            .len = param1.len,
            .data = param1.ptr,
        };
        return qtc.KStyleExtensions_CustomStyleHint(param1_str, @ptrCast(param2));
    }

    /// [Qt documentation](https://api-staging.kde.org/kstyleextensions.html#customSubElement)
    ///
    /// ``` param1: []const u8, param2: QtC.QWidget ```
    ///
    /// Returns: ``` qstyle_enums.SubElement ```
    pub fn CustomSubElement(param1: []const u8, param2: ?*anyopaque) i64 {
        const param1_str = qtc.libqt_string{
            .len = param1.len,
            .data = param1.ptr,
        };
        return qtc.KStyleExtensions_CustomSubElement(param1_str, @ptrCast(param2));
    }
};

File: src/extras-kwidgetsaddons/libkstyleextensions.cpp
#include <KStyleExtensions>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QWidget>
#include <kstyleextensions.h>
#include "libkstyleextensions.h"
#include "libkstyleextensions.hxx"

int KStyleExtensions_CustomControlElement(const libqt_string param1, const QWidget* param2) {
    QString param1_QString = QString::fromUtf8(param1.data, param1.len);
    return static_cast<int>(KStyleExtensions::customControlElement(param1_QString, param2));
}

int KStyleExtensions_CustomStyleHint(const libqt_string param1, const QWidget* param2) {
    QString param1_QString = QString::fromUtf8(param1.data, param1.len);
    return static_cast<int>(KStyleExtensions::customStyleHint(param1_QString, param2));
}

int KStyleExtensions_CustomSubElement(const libqt_string param1, const QWidget* param2) {
    QString param1_QString = QString::fromUtf8(param1.data, param1.len);
    return static_cast<int>(KStyleExtensions::customSubElement(param1_QString, param2));
}
